examples/knative/powershell/kn-ps-slack-vsphere-alarm/handler.ps1

```diff
@@ -0,0 +1,146 @@
+Function Process-Init {
+   Write-Host "$(Get-Date) - Processing Init`n"
+
+   Write-Host "$(Get-Date) - Init Processing Completed`n"
+}
+
+Function Process-Shutdown {
+   Write-Host "$(Get-Date) - Processing Shutdown`n"
+
+   Write-Host "$(Get-Date) - Shutdown Processing Completed`n"
+}
+
+Function Process-Handler {
+   param(
+      [Parameter(Position=0,Mandatory=$true)][CloudNative.CloudEvents.CloudEvent]$CloudEvent
+   )
+
+   # Decode CloudEvent
+   try {
+      $cloudEventData = $cloudEvent | Read-CloudEventJsonData -Depth 10
+   } catch {
+      throw "`nPayload must be JSON encoded"
+   }
+
+   if(${env:FUNCTION_DEBUG} -eq "true") {
+      Write-Host "$(Get-Date) - DEBUG: K8s Secrets:`n${env:SLACK_SECRET}`n"
+
+      Write-Host "$(Get-Date) - DEBUG: CloudEventData`n $(${cloudEventData} | Out-String)`n"
+   }
+
+   # Check the CloudEvent envelope properties to make sure this is vSphere Alarm Event AND it is our enriched vSphere Alarm
+   if($CloudEvent.type -eq "com.vmware.event.router/event.AlarmInfo" -and $CloudEvent.Subject -eq "AlarmStatusChangedEvent") {
+
+      if($cloudEventData.To -eq "red" -or $cloudEventData.To -eq "yellow") {
+
+         try {
+            $jsonSecrets = ${env:SLACK_SECRET} | ConvertFrom-Json
+         } catch {
+            throw "`nK8s secrets `$env:SLACK_SECRET does not look to be defined"
+         }
+
+         # Retrieve the vSphere Object Type
+         # Additional vSphere Types can be mapped to custom Slack icons
+         switch($cloudEventData.Entity.Entity.Type) {
+            "datastore" {
+               $objectType = ":database-6178:"
```

Comment:
is this a common slack symbol (available across all slack instances)?

Reply:
No, this was used for our specific example. If its not available, it'll simply show blank icon and I'd expect users to customize (locally) based on their needs